backend/backend/core/chains.py

```diff
@@ -1,17 +1,19 @@
 
+from langchain_core.output_parsers import PydanticOutputParser
 from langchain_core.prompts import ChatPromptTemplate
 from langchain_core.output_parsers import StrOutputParser
-
 from utils.logger import logger
 
 from .models import ModelRegistry
 from .tools import ToolRegistry
 
+from pydantic import BaseModel
```

Comment:
The line `model = model` does not have any effect. It seems like the intention might be to assign the passed `model` parameter to an instance variable, in which case it should be `self.model = model`. However, since this is not causing a bug based on the provided context, I am only pointing out the potential issue.